File: 226_invert_binary_tree/226_invert_binary_tree.java
class Solution {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode() {
        }

        TreeNode(int val) {
            this.val = val;
        }

        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.right = right;
            this.left = left;
        }
    }

    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return null;
        }
        TreeNode tmpRightNode = root.right;
        root.right = invertTree(root.left);
        root.left = invertTree(tmpRightNode);
        return root;
    }
}

File: 226_invert_binary_tree/226_invert_binary_tree.go
package invertbinarytree

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return nil
	}
	tmpRightNode := root.Right
	root.Right = invertTree(root.Left)
	root.Left = invertTree(tmpRightNode)
	return root
}
